### ISSUE TITLE
Proper setup for current anti-spam fee, setup target address and minimum in genesis file

### ISSUE THREAD

**Issue by ethanfrey** [iov-one/weave#264]
Current implementation of fee decorator:
https://github.com/iov-one/weave/blob/master/x/cash/decorator.go

Currently setup with hard-coded value (of nothing)
https://github.com/iov-one/weave/blob/master/cmd/bnsd/app/init.go#L78
https://github.com/iov-one/weave/blob/master/cmd/bnsd/app/app.go#L55

At the very least, the arguments passed in init.go, line 78 above should be definable in the genesis file. Especially the address that collects all fees.

We don't handle dynamic fees or anything at this point (other issue), just make current implementation usable/configurable as a first step

Consider building some more generic configuration options and resolving #202 while working on this issue (they seem to go hand-in-hand)

Also, this seems like a good time to pull fees outside of cash into it's own module before we start adding complexity. (but that is optional in resolving this issue)

**Comment by ethanfrey** [comment]
Note that this minimum fee will be enforced both on CheckTx and DeliverTx. (But validators can add other subjective criteria on CheckTx on a later story).

Also, we need a way for a client to query this minimal fee, to easily pay the minimum.

**Comment by ethanfrey** [comment]
@husio please look at this story before sprint planning and add any questions you have (eg. code level)

**Comment by husio** [comment]
I think I understand the current implementation, both the code and functionality.

Current implementation is setting single price for any operation. Processing a transaction cost X and the X is deducted from the main signer's account.
1. Should the fee work for certain transactions only? For example, we do not want to charge a fee for validators change transaction.
2. Should the fee be defined per transaction type? For example, we want to charge more for escrow creation than release. We also want to charge more for the batch message, but still give a discount because it's a batch processing.

**Comment by ethanfrey** [comment]
This current implementation is correct as *anti-spam*, to prevent people from filling up blocks, and should be something minimal (like 0.01 IOV). This maintains same behavior, just makes it usable and tuneable.

#232 will address your point 2 about per-transaction type "product fees".

We need to create another issue to allow a validator-subjective filter in CheckTx only, as a rapid response scenario to avoid spam (this means they will not gossip or propose it, but will process a block another validator creates).

All functionality changes mentioned above are not part of this issue
